crates/bevy_render/src/mesh/shape/icosphere.rs

```diff
@@ -0,0 +1,79 @@
+use hexasphere::shapes::IcoSphere;
+
+use crate::{
+    mesh::{Indices, Mesh},
+    pipeline::PrimitiveTopology,
+};
+
+/// A sphere made from a subdivided Icosahedron.
+#[derive(Debug)]
+pub struct Icosphere {
+    /// The radius of the sphere.
+    pub radius: f32,
+    /// The number of subdivisions applied.
+    pub subdivisions: usize,
+}
+
+impl Default for Icosphere {
+    fn default() -> Self {
+        Self {
+            radius: 1.0,
+            subdivisions: 5,
+        }
+    }
+}
+
+impl From<Icosphere> for Mesh {
+    fn from(sphere: Icosphere) -> Self {
+        if sphere.subdivisions >= 80 {
+            // https://oeis.org/A005901
```

Comment:
It looks like oeis content is available under the Creative Commons Non-Commerical 3.0 license: https://creativecommons.org/licenses/by-nc/3.0/

This is unfortunately not compatible with the MIT license (or the general concept of selling games).

Reply:
the `Icosphere` code is not modified in this commit, only extracted into an own file.
It looks like [this commit](https://github.com/bevyengine/bevy/commit/3da653e4dd74671a6ea62f22a35ee24eed4644f4) introduces the comment.

Reply:
oof we should remove that then :smile: 

@OptimisticPeach  any thoughts on this?

Reply:
Hmm, I think it'd be preferable to keep some kind of explanation there, however it's up to your discretion.

I'm not very keen on license rules, however might wikipedia be a better resource? https://en.wikipedia.org/wiki/Geodesic_polyhedron#Elements

Reply:
Okay, I drafted up a desmos graph to explain a method of calculating it:
https://www.desmos.com/calculator/8bbvlpkecj
However, it doesn't seem that desmos can be used for commercial applications or anything of the sort, hence, I made a gist with the same contents:
https://gist.github.com/OptimisticPeach/0851820de5694850d10f8b050beed125
I believe that linking to that, **and** additionally changing the code to reflect it, to the following:
```rs
let temp = sphere.subdivisions + 1;
let number_of_resulting_points = temp * temp * 10 + 2;
```
To better reflect these calculations.

Reply:
I changed the code and included @OptimisticPeach's explanation (in my opinion it is short enough to put it in the code). I think this PR should only contain MIT licensed code now.

Reply:
Sounds good to me. Thanks for resolving this so quickly!